src/__tests__/custom-rules/custom-rules.spec.ts

```diff
@@ -0,0 +1,55 @@
+import "../../custom-rules";
+import { jsonToSchema } from "../../schema-generator";
+import { TestHelper } from "../../utils";
+
+const ERROR_MESSAGE = "test error message";
+
+describe("custom-rules", () => {
+	it.each`
```

Comment:
I have added the unit tests for string and number, for arrays, the tests are added in the `fields` branch because there are no fields in branch that can do arrays: https://github.com/LifeSG/validation-schema-generator/blob/fields/src/__tests__/custom-rules/custom-rules.spec.ts